backend/dataall/modules/s3_datasets_shares/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -77,122 +77,149 @@ def check_s3_iam_access(self) -> None:
         kms_key_id = kms_client.get_key_id(key_alias)
 
         share_policy_service = S3SharePolicyService(
-            environmentUri=self.target_environment.environmentUri,
+            role_name=self.target_requester_IAMRoleName,
             account=self.target_environment.AwsAccountId,
             region=self.target_environment.region,
-            role_name=self.target_requester_IAMRoleName,
+            environmentUri=self.target_environment.environmentUri,
             resource_prefix=self.target_environment.resourcePrefix,
         )
-        share_resource_policy_name = share_policy_service.generate_policy_name()
 
-        if not share_policy_service.check_if_policy_exists():
-            logger.info(f'IAM Policy {share_resource_policy_name} does not exist')
-            self.bucket_errors.append(ShareErrorFormatter.dne_error_msg('IAM Policy', share_resource_policy_name))
-            return
+        # Parses all policy documents and extracts s3 and kms statements
+        share_policy_service.initialize_statements()
 
-        if not share_policy_service.check_if_policy_attached():
-            logger.info(
-                f'IAM Policy {share_resource_policy_name} exists but is not attached to role {self.share.principalRoleName}'
+        share_resource_policy_name = share_policy_service.generate_indexed_policy_name(index=0)
+        is_managed_policies_exists = True if share_policy_service.get_managed_policies() else False
+
+        if not is_managed_policies_exists:
+            warn(
+                "Convert all your share's requestor policies to managed policies with indexes. Deprecation >= ?? ",
+                DeprecationWarning,
+                stacklevel=2,
             )
-            self.bucket_errors.append(
-                ShareErrorFormatter.dne_error_msg('IAM Policy attached', share_resource_policy_name)
+            old_managed_policy_name = share_policy_service.generate_old_policy_name()
+            if not share_policy_service.check_if_policy_exists(policy_name=old_managed_policy_name):
```

Comment:
Same comments as in the access_points_manager: https://github.com/data-dot-all/dataall/pull/1650#discussion_r1822289919 and https://github.com/data-dot-all/dataall/pull/1650#discussion_r1822300141